SalesforceCommerceCloud/pwa-kit
# Default locale in multisite setup

# Description

A small PR to address the concept of default locale in a multisite setup:

- For `IntlProvider`, set the default locale to be the locale of the default messages ← not the same as default locale in Business Manager
- When loading a translation file and it can't be found, we'll load an empty object so that react-intl would fall back to the inline default messages ← this helps during development, when some translation files would be missing and would come later from the translators
- Some code cleanup

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`



# How to Test-Drive This PR

1. Visit http://localhost:3000/us/en-CA/
2. Verify that the page loads fine, even though the en-CA translation does not actually exist yet.
2. In your browser console, you'd see a warning like `Missing message: "footer.link.site_map" for locale "en-CA", using default message as fallback.`
3. In your dev server console, you'd see `Error: Cannot find module './en-CA.json'`

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
